### ISSUE TITLE
Reporting wrong disk space

### ISSUE THREAD

**Issue by andrefreitas** [rancher/rancher#2023]
We have provisioned hosts in Linode with Ubuntu 14.04.1 LTS (GNU/Linux 4.1.5-x86_64-linode61 x86_64). Rancher is reporting a wrong value for the disk space (actually is increasing every day).

# Example
All the servers have 200GB of real space and the reported space is different.

![screen shot 2015-09-11 at 09 41 09](https://cloud.githubusercontent.com/assets/1192089/9810593/c14a66c6-5869-11e5-8963-fc0be323a2dd.png)

**Comment by cloudnautique** [comment]
@andrefreitas if you wouldn't mind sharing, it would be helpful to have the API info. Could you go to 'View in API' on a couple of hosts and paste the Info section? Feel free to sanitize as needed. I'm curious about the diskInfo section primarily.

**Comment by andrefreitas** [comment]
@cloudnautique sure, there is an example:

```json
{
"id": "1h1",
"type": "host",
"links": {
"self": "…/v1/projects/1a5/hosts/1h1",
"account": "…/v1/projects/1a5/hosts/1h1/account",
"clusters": "…/v1/projects/1a5/hosts/1h1/clusters",
"containerEvents": "…/v1/projects/1a5/hosts/1h1/containerevents",
"hostLabels": "…/v1/projects/1a5/hosts/1h1/hostlabels",
"instances": "…/v1/projects/1a5/hosts/1h1/instances",
"ipAddresses": "…/v1/projects/1a5/hosts/1h1/ipaddresses",
"loadBalancerHostMaps": "…/v1/projects/1a5/hosts/1h1/loadbalancerhostmaps",
"loadBalancers": "…/v1/projects/1a5/hosts/1h1/loadbalancers",
"physicalHost": "…/v1/projects/1a5/hosts/1h1/physicalhost",
"serviceEvents": "…/v1/projects/1a5/hosts/1h1/serviceevents",
"storagePools": "…/v1/projects/1a5/hosts/1h1/storagepools",
"stats": "…/v1/projects/1a5/hosts/1h1/stats",
"hostStats": "…/v1/projects/1a5/hosts/1h1/hoststats",
"containerStats": "…/v1/projects/1a5/hosts/1h1/containerstats",
},
"actions": {
"update": "…/v1/projects/1a5/hosts/1h1/?action=update",
"deactivate": "…/v1/projects/1a5/hosts/1h1/?action=deactivate",
},
"name": "host1",
"state": "active",
"accountId": "1a5",
"agentState": null,
"computeTotal": 1000000,
"created": "2015-09-09T16:03:54Z",
"createdTS": 1441814634000,
"description": null,
"info": {
"osInfo": {
"versionDescription": "trusty",
"kernelVersion": "4.1.5-x86_64-linode61",
"distribution": "Ubuntu",
"version": "14.04",
"dockerVersion": "Docker version 1.8.1, build d12ea79",
},
"cpuInfo": {
"count": 4,
"cpuCoresPercentages": [ 4 items
5.87,
4.582,
4.124,
13.138,
],
"loadAvg": [ 3 items
0.53,
1.15,
1.05,
],
"mhz": 2499.996,
"modelName": "Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz",
},
"memoryInfo": {
"memTotal": 3950.324,
"swapTotal": 255.996,
"cached": 804.434,
"swapCached": 5.078,
"swapFree": 72.961,
"memAvailable": 1594.594,
"memFree": 600.711,
"inactive": 1512.566,
"active": 1552.105,
"buffers": 134.355,
},
"diskInfo": {
"mountPoints": {
"/dev/root": {
"total": 96445.605,
"free": 82674.852,
"used": 13770.754,
"percentUsed": 14.28,
},
"/dev/mapper/docker-8:0-81963-38c1e0bbd0465e0b03e60c4b8c706c3ff4b5615be92c311392d3289f8e78488e": {
"total": 100665.281,
"free": 100283.328,
"used": 381.953,
"percentUsed": 0.38,
},
"/dev/mapper/docker-8:0-81963-05d5f6771a7fc334ed8fb6c76d68ea560dc197ccbe3bfd15be98e4caafd6b66e": {
"total": 100665.281,
"free": 100139.199,
"used": 526.082,
"percentUsed": 0.52,
},
},
},
},
"kind": "docker",
"labels": {
"sanitized"
},
"physicalHostId": "1ph1",
"removed": null,
"transitioning": "no",
"transitioningMessage": null,
"transitioningProgress": null,
"uuid": "5222a894-f4e4-49b7-9d99-a5b902975483",
```

**Comment by cloudnautique** [comment]
We have had trouble with accounting mapped devices. We get the stats from cAdvisor, and it looks like the newer versions have some easier ways to account for actual space. Will be working to address this.

I think we can also address #1586 at the same time.

**Comment by cloudnautique** [comment]
@andrefreitas, can you also provide the output of `docker info` and `docker version` please? I'd like to see which storage driver your using.

**Comment by andrefreitas** [comment]
@cloudnautique I don't have access now to the machines but we are using docker without any 3rd party storage drivers.

**Comment by cloudnautique** [comment]
Do you know if it is/was using overlayfs, Aufs or device mapper? Docker by default uses an internal storage driver independent of the third party volume drivers. I'm guessing it is device mapper, but I wanted to confirm.

**Comment by andrefreitas** [comment]
@cloudnautique  there goes the info:

    > docker info

```
Containers: 13
Images: 411
Storage Driver: devicemapper
 Pool Name: docker-8:0-81963-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 10.57 GB
 Data Space Total: 107.4 GB
 Data Space Available: 85.94 GB
 Metadata Space Used: 21.34 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.126 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.77 (2012-10-15)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.1.5-x86_64-linode61
Operating System: Ubuntu 14.04.1 LTS
CPUs: 4
Total Memory: 3.858 GiB
Name: host1.babel.school
ID: Y7BD:WKTL:2F2J:R4ZY:CBNH:4NJA:DZDR:QVS5:HY4W:ZCRJ:IPFY:DSVR
WARNING: No swap limit support
```

    docker version

```
Client:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:19:00 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:19:00 UTC 2015
 OS/Arch:      linux/amd64
```

Linode VPS don't have AUFS. If you need more information, please let me know.

**Comment by sangeethah** [comment]
Tested with build from master - Sep 18

Host had 10 GB disk space.

Set docker to use devicemapper by setting the following parameter in /etc/default/docker.

export DOCKER_DRIVER=devicemapper

Added host to rancher server with 10 gb disk.

Rancher server reports wrong disk space as 108 gb.

<img width="347" alt="screen shot 2015-09-21 at 10 19 04 am" src="https://cloud.githubusercontent.com/assets/4266958/9999283/5f1c43fa-604a-11e5-80e6-3fcdcfc642a0.png">


root@sangee-testdevmapper:/var/log/rancher# docker info
Containers: 5
Images: 17
Storage Driver: devicemapper
 Pool Name: docker-8:1-416263-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.286 GB
 Data Space Total: 107.4 GB
 Data Space Available: 5.255 GB
 Metadata Space Used: 2.744 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.77 (2012-10-15)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-28-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 1
Total Memory: 3.614 GiB
Name: sangee-testdevmapper
ID: TA5O:HGYN:RFEP:M44T:QZST:XFUL:OFSJ:WDN6:JIF2:CPLZ:WCRO:6OKK
WARNING: No swap limit support
root@sangee-testdevmapper:/var/log/rancher# 
```

**Comment by sangeethah** [comment]
Tested with server version - v0.39.0-rc1

Host had an Ubuntu server with 10 GB disk space.

Set docker to use devicemapper by setting the following parameter in /etc/default/docker.

export DOCKER_DRIVER=devicemapper

Added host to rancher server with 10 gb disk.

Rancher server reports correct disk space - reports 9.9 GiB.
